### ISSUE TITLE
Some interface design decisions regarding input data, outcome in data2draws

### ISSUE THREAD

**Issue by Dashadower** [hyunjimoon/DataInDM#22]
A rough implementation of data2draws is partly implemented. You can write the following python code:
```
model = StanVensimModel("ds_white_sterman", am, list(range(0, 10)))  # range(0, 10) is the integrations times
model.set_prior("inventory_adjustment_time", "normal", 0, 1)
model.set_prior("minimum_order_processing_time", "normal", 0, 1)
```

This (currently) results in the following generated code:
```
#include ds_white_sterman_functions.stan

data{
}

transformed data{
    int T = 10;
    array[T] real times = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
}

transformed parameters {
    # Initial ODE values
    real inventory_initial = 2 + 2 * 10000;
    real expected_order_rate_initial = 10000;
    real work_in_process_inventory_initial = 8 * max(0,10000 + 2 + 2 * 10000 - 2 + 2 * 10000 / 8);

    vector[3] initial_outcome;  # Initial ODE state vector
    initial_outcome[1] = inventory;
    initial_outcome[2] = expected_order_rate;
    initial_outcome[3] = work_in_process_inventory;

    vector[3] integrated_result[T] = ode_rk45(vensim_ode, initial_outcome, initial_time, times, minimum_order_processing_time, inventory_adjustment_time);
}

parameters{
    real inventory_adjustment_time;
    real minimum_order_processing_time;
}

model{
    inventory_adjustment_time ~ normal(0, 1);
    minimum_order_processing_time ~ normal(0, 1);
}
```

The problem is how we're going to expose `vector[3] integrated_result[T]` to the user. There are 2 ways the user would want to access it. 1. along the time axis(which results in a array of length `T` with a single stock) 2. for a single timestep(which results in a vector of length 3)

Another problem is linking the codegen with data specified by the user. I'm not sure what form the data would have to be inputted from the user. If data was specified, we'd have to construct the data block that matches the inputted data.

**Comment by hyunjimoon** [comment]
Some default decisions, with the help of precious prior knowledge on SD modeler's default decisions :) 

1. penalize only the states (=stock)
all comparison between simulated and observed state is done with stock variable. Users should change the structure within vensim, by adding artificial stock ([var_name]_state) that ensures its value is close to the value of the original type (observed data of flow or auxiliary variable).

2. `state_init` 
`state_init` is `ass_param`; when user select to include stock variable as `est_param`, we add two: initial (latent) value  `z_init` and measurement error scale `z_sigma`

3. `times`
When `state_init` is `ass_param`, its value being the first value of the observation, the length of `times` should be 19.
This may differ if `state_init` is `est_param` as then the penalizing (between `z_init` and `y_init` happens from time 0. To be specific, `times` is  1:19 for `state_init` is `ass_param` and  0:19 `state_init` is `est_param` i.e. 1 to n_t as we don't want to simulated what's already known as explained [here](https://discourse.mc-stan.org/t/issues-getting-non-linear-ode-model-inference-going-starting-off-with-ode-ckrk-function-error-message/25787/2?u=hyunji.moon). 

4. `data2draws` data block has the following components 
```
    int n_obs_state;
    int n_t;
    real initial_time;
    array[n_t] real times;
    array[n_t] vector[n_obs_state] y; //observed states
```

5. `draws2data` data block has the following components
```
    int n_obs_state;
    int n_t;
    real initial_time;
    array[n_t] real times;
    array[n_t] vector[n_obs_state] y; 
```

**Comment by Dashadower** [comment]
> I don't know whether times in data vs transformed data is better.

I would rephrase that question into, "Do you expect the integration times to be able to vary, without **any** changes to the existing model?". If the answer is yes, then I would believe that moving times into data would be the correct option.

However, since we also handle the input data automatically within pysd, it would't hurt to just move it into data regardless, since that would remove the need for recompilation in the event that times changes.
